### PR TITLE
configure: save CFLAGS when testing for valid flags from `debug_c_other_flags`

### LINKED ISSUE #4742: v1.7.0: configure fails with `--enable-debug`
Hi,

Configure fails on our CentOS 6 system when using the `--enable-debug` with the error
```
configure: error: dlopen not found.  libfabric requires libdl.
```
This is because it has ended up with `-fstack-protector-strong` in `CFLAGS` even though it's not supported by our GCC version; all tests after this point the fail until it reaches the libdl test which is fatal.

It looks like it does test for `-fstack-protector-strong` and correctly detect that it's unsupported, but then leaves that flag in `CFLAGS`. I think commit 7bb6649ad6fb7bedaf5d387d68e4266e0d00feed is the culprit here; I'll patch it and PR.

Cheers,
Ben

### PR DESCRIPTION
**Pull request by benmenadue**
This fixes #4742.

Signed-off-by: Ben Menadue <ben.menadue@nci.org.au>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -108,7 +109,8 @@ AS_IF([test x"$enable_debug" != x"no"],
        debug_c_other_flags=$good_flags
        unset good_flags
 
-       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} ${debug_c_other_flags} $CFLAGS"],
+       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} ${debug_c_other_flags} ${CFLAGS_save}"]
+       unset CFLAGS_save,
```
Comment (jsquyres) [inline configure.ac]: Need to move the `]` from the end of the `CFLAGS="-g ...` line to the `unset ...` line (before the `,`).

I.e.

```
       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} ${debug_c_other_flags} ${CFLAGS_save}"
       unset CFLAGS_save],
```

Comment (benmenadue) [reply configure.ac]: Ah, yep, thanks, fixed. I'm surprised that it still built fine for me with that.

Comment (jsquyres) [reply configure.ac]: m4 is a fickle beast.

my-gosh

**Comment by shefty** [comment]
thanks!  Based on the issue, I'll port this to 1.7.x branch as well.

**Pull request merged**